### PR TITLE
Raised value error on incorrect na_option

### LINKED ISSUE #19499: Rank 'na_option="bottom"' Usage Clarification
#### Code Sample, a copy-pastable example if possible

```python
df = pd.DataFrame({'val': [2, np.nan, 2, 8, 2, np.nan, 6]})

# Works as documented - missing values are highest rank
In []: df.rank(na_option='top')
Out []: 
   val
0  4.0
1  1.5
2  4.0
3  7.0
4  4.0
5  1.5
6  6.0

# Technically works - missing values are lowest rank
In []: df.rank(na_option='bottom')
Out []: 
   val
0  2.0
1  6.5
2  2.0
3  5.0
4  2.0
5  6.5
6  4.0 

# However, we could say anything besides 'foo'
In []: df.rank(na_option='foo')
Out []: 
   val
0  2.0
1  6.5
2  2.0
3  5.0
4  2.0
5  6.5
6  4.0 
```
#### Problem description

For the sake of being explicit it would be better to raise for an unknown `na_option`, or alternately update the documentation to reflect that any value outside of 'keep' and 'top' would trigger this behavior

<details>

INSTALLED VERSIONS
------------------
commit: d3f7d2a666aa824e2df98083aa5c1fd9bb63252e
python: 3.6.3.final.0
python-bits: 64
OS: Darwin
OS-release: 17.4.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.0.dev0+169.gd3f7d2a66.dirty
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: 0.26.1
numpy: 1.13.3
scipy: 1.0.0
pyarrow: 0.8.0
xarray: 0.10.0
IPython: 6.2.1
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: 0.4.0
matplotlib: 2.1.1
openpyxl: 2.5.0b1
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.1.13
pymysql: 0.7.11.None
psycopg2: None
jinja2: 2.10
s3fs: 0.1.2
fastparquet: 0.1.3
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by raguiar2**
- [X] closes #19499
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Just need test(s) to ensure the behavior. Can also add a whatsnew for 0.24

**Comment by WillAyd** [comment]
Also can you make sure this works for all of `Series`, `DataFrame` and `GroupBy` objects? Behavior should be identical across all three

**Comment by raguiar2** [comment]
Not sure why travis-ci is failing. Looks unrelated/might have to do with an upstream dependency? Do I have to create a new PR or can I re-run the build somehow?

File: pandas/tests/frame/test_rank.py [hunk 1]
```diff
@@ -191,6 +191,20 @@ def test_rank_na_option(self):
         tm.assert_numpy_array_equal(ranks0.values, exp0)
         tm.assert_numpy_array_equal(ranks1.values, exp1)
 
+        # bad values throw error
+        msg = "na_option must be one of 'keep', 'top', or 'bottom'"
+
+        with pytest.raises(ValueError) as excinfo:
+            self.frame.rank(na_option='bad', ascending=False)
+
+        assert msg == str(excinfo.value)
```
Comment (WillAyd) [inline pandas/tests/frame/test_rank.py]: Instead of doing this you can pass the expected `msg` as an argument to `tm.assert_raises_regex`

**Review by WillAyd — CHANGES_REQUESTED** [review]
One comment (applicable to all tests) but otherwise LGTM. Travis failures look unrelated so will see on re-push if resolved, but otherwise I don't think it's a blocker for this PR

**Comment by raguiar2** [comment]
@WillAyd PTAL!

**Review by WillAyd — APPROVED** [review]
This lgtm - @jreback any feedback before I merge?

**Comment by jreback** [comment]
lgtm

do we have an open issue for this?

**Comment by WillAyd** [comment]
Updated original comment to make linkage clearer. Merging later today if no objections

**Comment by TomAugspurger** [comment]
Thanks @raguiar2!

**Pull request merged**